src/main/java/org/killbill/billing/plugin/core/resources/PluginHealthcheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2014-2017 Groupon, Inc
+ * Copyright 2014-2017 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.core.resources;
+
+import java.util.Dictionary;
+
+import javax.annotation.Nullable;
+
+import org.jooby.MediaType;
+import org.jooby.Result;
+import org.jooby.Results;
+import org.jooby.Status;
+import org.killbill.billing.plugin.service.Healthcheck;
+import org.killbill.billing.plugin.service.Healthcheck.HealthStatus;
+import org.killbill.billing.tenant.api.Tenant;
+
+// Specified in the concrete class
+//@Singleton
+//@Path("/healthcheck")
+public abstract class PluginHealthcheck {
```

Comment:
1. It doesn't have to be `abstract`, but it's just to show the developer that it is meant to be implemented in the plugin (that base class just shares some utilities). My first attempt was actually to do the opposite and have the servlet here (hence the commented out annotations), and have a single `protected` method that the plugin would implement. Unfortunately, I couldn't make it work: jooby inheritance somehow makes the concrete class not visible (maybe because annotations aren't inherited?). Happy to revisit if @andrenpaes has an idea on how to solve this?
2. Eventually, I think it would be indeed a nice to have: the core endpoint would iterate through all healthchecks and gather a high-level view of the health of the system. For now, it can just be exposed as an OSGI service, so that company-specific healthchecks (i.e. defined in plugins, instead of using the native one) can do precisely this.
3. It is exactly done that way (see Adyen PR) 😃 I've just added on top of it a servlet that calls that service for convenience.